### PR TITLE
Set up `release/3.1.x` branch

### PR DESCRIPTION
**Pull request by khieta**
*Description of changes:*

Setup for `release/3.1.x` branch. This should resolve upstream CI failures. Also updated CI to use the relevant branch of `cedar-policy` rather than `main`.

### CODE REVIEW

**Comment by khieta** [comment]
Converting to draft until I fix build failures.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -56,21 +56,21 @@ jobs:
     needs: get-branch-name
     uses: ./.github/workflows/run_example_use_cases_reusable.yml
     with:
-      cedar_policy_ref: "refs/heads/main" # use the latest commit on main
-      cedar_examples_ref: "${{ github.head_ref }}" # use the current PR's commit
+      cedar_policy_ref: refs/heads/${{ needs.get-branch-name.outputs.branch_name }}
```
Comment (khieta) [inline .github/workflows/ci.yml:59]: If this is too hacky, I could also switch to the hardcoded "refs/heads/release/3.1.x".

**Review by john-h-kastner-aws — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (john-h-kastner-aws) [reply .github/workflows/ci.yml:59]: either way seems fine. I wouldn't object to hardcoding, but this should mean we wouldn't need to upadate the CI on this branch if we make a 3.2.x, right?

Comment (khieta) [reply .github/workflows/ci.yml:59]: Yes, if I hardcode this, we'll need to update each release branch manually. But that should be easy to maintain -- just `ctrl-f` for the version you want to change.

**Review by cdisselkoen — APPROVED** [review]

**Pull request merged**